### PR TITLE
Add isPresent to ValueIn interface, 

### LINKED ISSUE #378: There's no way to tell if a value was present
Currently when reading from a `WireIn`, if you call
```
ValueIn valueIn = wireIn.read("fieldName");
```

the only way to determine if the field was present is
```
boolean wasPresent = valueIn instanceof DefaultValueIn;
```

It feels like there should be a way to do this via the `ValueIn` interface, rather than reaching under the hood in this way and relying on implementation details.

### Proposed solution
My proposed solution is to add a method `boolean isPresent()` to the `ValueIn` interface. `DefaultValueIn` could return `false` for this and every `ValueIn` that represented a present value could return `true`.

### PR DESCRIPTION
**Pull request by nicktindall**
also replace occurrences of `xxx instanceof DefaultValueIn`, Fixes #378

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/wire/WireMarshaller.java [hunk 1]
```diff
@@ -755,9 +751,10 @@ protected void getValue(Object o, @NotNull ValueOut write, Object previous) {
         @Override
         protected void setValue(Object o, @NotNull ValueIn read, boolean overwrite) {
             @NotNull StringBuilder sb = unsafeGetObject(o, offset);
-            if (sb == null)
+            if (sb == null) {
                 sb = new StringBuilder();
                 unsafePutObject(o, offset, sb);
+            }
```
Comment (nicktindall) [inline src/main/java/net/openhft/chronicle/wire/WireMarshaller.java:757]: I noticed this when I auto-formatted. The indenting seemed to match the intention, but the lack of brackets meant we always `unsafePut..` the string buffer, even if `unsafeGet..` returned non-null.

Not sure if this was intentional for some reason?

Comment (minborg) [reply src/main/java/net/openhft/chronicle/wire/WireMarshaller.java:757]: We should always use brackets in Java.

File: src/main/java/net/openhft/chronicle/wire/ValueIn.java [hunk 2]
```diff
@@ -548,6 +548,15 @@ default Object typePrefixOrObject(Class tClass) {
     @Nullable
     Object objectWithInferredType(Object using, SerializationStrategy strategy, Class type);
 
+    /**
+     * Is the value present?
```
Comment (minborg) [inline src/main/java/net/openhft/chronicle/wire/ValueIn.java]: Returns if a value is present.

File: src/main/java/net/openhft/chronicle/wire/WireMarshaller.java [hunk 3]
```diff
@@ -554,18 +554,16 @@ protected void copy(Object from, Object to) throws IllegalAccessException {
         protected abstract void getValue(Object o, ValueOut write, Object previous) throws IllegalAccessException;
 
         protected void readValue(Object o, Object defaults, ValueIn read, boolean overwrite) throws IllegalAccessException {
-            if (read instanceof DefaultValueIn) {
+            if (!read.isPresent()) {
```
Comment (minborg) [inline src/main/java/net/openhft/chronicle/wire/WireMarshaller.java:557]: This is much better!

**Review by minborg — APPROVED** [review]

**Pull request merged**